clang/lib/AST/PreorderAST.cpp

```diff
@@ -403,51 +403,62 @@ bool PreorderAST::GetDerefOffset(Node *UpperNode, Node *DerefNode,
   return true;
 }
 
-bool PreorderAST::IsEqual(Node *N1, Node *N2) {
+Result PreorderAST::Compare(Node *N1, Node *N2) {
   // If both the nodes are null.
   if (!N1 && !N2)
-    return true;
+    return Result::Equal;
 
   // If only one of the nodes is null.
-  if ((N1 && !N2) || (!N1 && N2))
-    return false;
+  if (!N1 && N2)
+    return Result::LessThan;
+  if (N1 && !N2)
+    return Result::GreaterThan;
 
   if (const auto *O1 = dyn_cast<OperatorNode>(N1)) {
-    // If the types of the nodes mismatch.
+    // OperatorNode > LeafNode.
     if (!isa<OperatorNode>(N2))
-      return false;
+      return Result::GreaterThan;
 
     const auto *O2 = dyn_cast<OperatorNode>(N2);
 
     // If the Opcodes mismatch.
-    if (O1->Opc != O2->Opc)
-      return false;
+    if (O1->Opc < O2->Opc)
+      return Result::LessThan;
+    if (O1->Opc > O2->Opc)
+      return Result::GreaterThan;
+
+    unsigned ChildCount1 = O1->Children.size(),
```

Comment:
`O1->Children` and `O2->Children` are `llvm::SmallVector`s and their sizes are of type `size_t`. So it's better to use `size_t` instead of `unsigned`.